src/common/databaseFilters.ts

```diff
@@ -0,0 +1,31 @@
+import { ServerSideFilterMethod } from "@/components/Tables/Filters";
+import { nullPostcodeDisplay } from "./format";
+import { DbClientRow, DbParcelRow } from "@/databaseUtils";
+
+export const fullNameMethodHelper = <DbData extends DbClientRow | DbParcelRow>(
+    columnLabel: Extract<keyof DbData, "full_name" | "client_full_name">
+): ServerSideFilterMethod<DbData, string> => {
+    return (query, state) => query.ilike(`${columnLabel}`, `%${state}%`);
+};
+
+export const postcodeMethodHelper = <DbData extends DbClientRow | DbParcelRow>(
+    columnLabel: Extract<keyof DbData, "address_postcode" | "client_address_postcode">
+): ServerSideFilterMethod<DbData, string> => {
+    return (query, state) => {
+        if (state === "") {
+            return query;
+        }
+        if (nullPostcodeDisplay.toLowerCase().includes(state.toLowerCase())) {
+            return query.or(
+                `${String(columnLabel)}.ilike.%${state}%, ${String(columnLabel)}.is.null`
```

Comment:
why do we need String here?

Reply:
thanks for noting this! an artifact from a previous version of the code